guardrails/validator_base.py

```diff
@@ -422,14 +422,23 @@ def __init__(
                     """,
                     FutureWarning,
                 )
+        self.on_fail_descriptor: str | OnFailAction = "custom"
```

Comment:
Why do we change the default behavior to be custom instead of NOOP??

Reply:
This doesn't change the custom behavior - NOOP gets picked up as an enum value and is set in line 428/429. Adding a test to ensure this is the case.

Reply:
resolved offline